test/suggestion.cpp

```diff
@@ -0,0 +1,196 @@
+ * NON-INFRINGEMENT.  See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ *
+ */
+
+#include <zim/zim.h>
+#include <zim/archive.h>
+#include <zim/search.h>
+#include <zim/writer/creator.h>
+#include <zim/writer/item.h>
+#include <zim/writer/contentProvider.h>
+
+#include "tools.h"
+
+#include "gtest/gtest.h"
+
+namespace {
+
+  class TestItem : public zim::writer::BasicItem {
+      public:
+      TestItem(const std::string& path, const std::string& mimetype, const std::string& title)
+        : BasicItem(path, mimetype, title) {}
+
+      virtual std::unique_ptr<zim::writer::ContentProvider> getContentProvider() const {
+        return std::unique_ptr<zim::writer::ContentProvider>(new zim::writer::StringProvider(""));
+      }
+  };
+
+  // Helper class to create a temporary directory and zim from titles list
+  // Remove the file along with the temporary directory once test is done.
+    class TempZimArchive : zim::unittests::TempFile {
+    public:
+      std::string tempZimPath;
+      TempZimArchive(const char* tempPath) : zim::unittests::TempFile {tempPath} {}
+
+      zim::Archive createZimFromTitles(std::vector<std::string> titles) {
+        tempZimPath = this->path() + ".zim";
+        zim::writer::Creator creator;
+        creator.configIndexing(true, "eng");
+        creator.startZimCreation(tempZimPath);
+
+        // add dummy items with given titles
+        for (auto title : titles) {
+          std::string path = "dummyPath" + title;
+          auto item = std::make_shared<TestItem>(path, "plain/text", title);
+          creator.addItem(item);
+        }
+        creator.addMetadata("Title", "This is a title");
+        creator.finishZimCreation();
+        return zim::Archive(tempZimPath);
+      }
+
+      ~TempZimArchive() {
+        remove(tempZimPath.c_str());
+      }
+  };
+
+  std::vector<std::string> getSuggestions(const zim::Archive archive, std::string query, int range) {
+    zim::Search search(archive);
+    search.set_suggestion_mode(true);
+    search.set_query(query);
+    search.set_range(0, range);
+    search.set_verbose(true);
+
+    std::vector<std::string> result;
+    for (auto entry = search.begin();entry!=search.end();entry++) {
+      std::cout<<(*entry).getTitle()<<entry.get_score()<<std::endl;
+      result.push_back((*entry).getTitle());
+    }
+
+    return result;
+  }
+
+  TEST(Suggestion, emptyQuery) {
+    std::vector<std::string> titles = {
+                                        "fooland",
+                                        "berlin wall",
+                                        "hotel berlin, berlin",
+                                        "again berlin",
+                                        "berlin",
+                                        "not berlin",
```

Comment:
Not bad, but we don’t need a trailing comma at the end of the list here and in the other lists.

Reply:
I personally like the comma at the end of a list (if the grammar allow it).
This way, when we add a new item (line) to the list, we don't have to change the previous line. It simplify the diff (and the review).

(It is just a comment, do not add it back).